src/main/java/com/pusher/client/channel/impl/PresenceChannelImpl.java

```diff
@@ -90,12 +138,14 @@ public String toSubscribeMessage() {
             final String json = GSON.toJson(jsonObject);
 
             return json;
-        }
-        catch (final Exception e) {
+        } catch (final Exception e) {
             throw new AuthorizationFailureException("Unable to parse response from Authorizer: " + authResponse, e);
         }
     }
 
+    private final Object lock = new Object();
+    private final Map<String, Set<PresenceChannelEventListener>> eventNameToListenerMap = new HashMap<String, Set<PresenceChannelEventListener>>();
+
     @Override
     public void bind(final String eventName, final SubscriptionEventListener listener) {
```

Comment:
`bind` method logic has not changed, but I had to create it's own `bind` method in `PresenceChannelImpl` as it's a dependency of `onMessage` method. I basically copied the logic from `ChannelsImpl` but changed the type from `SubscriptionEventListener` -> `PresenceChannelEventListener`